code/__HELPERS/_lists.dm

```diff
@@ -18,7 +18,6 @@
 #define LAZYFIND(L, V) L ? L.Find(V) : 0
 #define LAZYACCESS(L, I) (L ? (isnum_safe(I) ? (I > 0 && I <= length(L) ? L[I] : null) : L[I]) : null)
 #define LAZYSET(L, K, V) if(!L) { L = list(); } L[K] = V;
-#define LAZYLEN(L) length(L)
```

Comment:
iirc the reason this exists is so people are aware that the list youre touching is a lazy list, that way its easier to keep track. Yes the define is mostly redundant, but if someone sees `LAZYLEN`, they instantly know that what they are touching is a lazy list, and should use the other `LAZY` operators.

Reply:
I see lazylen in normal lists all the time, so that's not the case

Reply:
Thats due to improper usage

Reply:
Does it hurt to have this helper? It shouldn't affect performance at all.

Reply:
This is a fairly unnecessary change imo. I see the losses being higher than the gains, especially when it comes to porting code from codebases that still have it.

Reply:
> Does it hurt to have this helper? It shouldn't affect performance at all.

the concern is that it's confusing and almost all uses here haven't been correct. Also technically untrue because it affects compile time

> Thats due to improper usage

I've seen you request changes for the use of `LAZYLEN` over `.len` or `length()` multiple times where it hasn't been needed, so improper usage is a problem that is completely universal with this define

> especially when it comes to porting code from codebases that still have it.

This is a fair point, but not a big one. You're just one search and replace away from getting it to work.